fastfetch: additional feature flags; fastfetchMinimal: init at 2.42.0

khaneliman opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done
Closes https://github.com/NixOS/nixpkgs/issues/401717

Cross checking https://github.com/fastfetch-cli/fastfetch/wiki/Dependencies and https://github.com/fastfetch-cli/fastfetch/blob/dev/CMakeLists.txt to allow more granular control over the features available for a fastfetch build. Also adding a package alias for feature flags disabled to get a smaller closure for those who don't need a full fat install. 
<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

khaneliman committed:
fastfetch: add additional feature flags

Attempt to allow creating more customization options to reduce closure
size, for those who don't need all the features.

Aleksanaa commented on pkgs/by-name/fa/fastfetch/package.nix:
```diff
@@ -60,6 +61,7 @@ stdenv.mkDerivation (finalAttrs: {
   ];
 
   nativeBuildInputs = [
+    autoPatchelfHook
```
autoPatchelfHook doesn't exist on darwin. I don't know if there are equivalent

khaneliman replied:
Well, shit.

khaneliman committed:
fastfetch: refactor buildInputs

khaneliman committed:
fastfetch: add libsysprof-catpure for glib

Package 'sysprof-capture-4', required by 'glib-2.0', not found
Package sysprof-capture-4 was not found in the pkg-config search path.
fastfetch> Perhaps you should add the directory containing
`sysprof-capture-4.pc'
fastfetch> to the PKG_CONFIG_PATH environment variable

khaneliman committed:
fastfetchMinimal: init at 2.42.0

Fasfetch package with all the feature flags disabled.

isabelroses commented on pkgs/top-level/all-packages.nix:
```diff
@@ -1868,6 +1868,19 @@ with pkgs;
 
   libfx2 = with python3Packages; toPythonApplication fx2;
 
+  fastfetchMinimal = fastfetch.override {
+    audioSupport = false;
+    flashfetchSupport = false;
+    gnomeSupport = false;
+    imageSupport = false;
+    openclSupport = false;
+    rpmSupport = false;
+    sqliteSupport = false;
+    vulkanSupport = false;
+    waylandSupport = false;
+    x11Support = false;
+  };
```
NIT: Instead of crowding `all-packages.nix` maybe you could do something similar to `fishMinimal` https://github.com/NixOS/nixpkgs/pull/387070

khaneliman replied:
Yeah I like that more. Will change when I get back to my computer

khaneliman commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 403209`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>fastfetch</li>
    <li>fastfetch.man</li>
    <li>fastfetchMinimal</li>
    <li>fastfetchMinimal.man</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>fastfetch</li>
    <li>fastfetch.man</li>
    <li>fastfetchMinimal</li>
    <li>fastfetchMinimal.man</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>fastfetch</li>
    <li>fastfetch.man</li>
    <li>fastfetchMinimal</li>
    <li>fastfetchMinimal.man</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>fastfetch</li>
    <li>fastfetch.man</li>
    <li>fastfetchMinimal</li>
    <li>fastfetchMinimal.man</li>
  </ul>
</details>
